### ISSUE TITLE
Stale mounts are left if pod is deleted before nodePublish is completed

### ISSUE THREAD

**Issue by raunakkumar** [kubernetes/kubernetes#89112]
When a pod is deleted before nodePublish is done, but after nodeStage is done, it will not be unmounted by kubelet. This causes stale mounts on host to be left around.

**Comment by msau42** [comment]
@kubernetes/sig-storage-bugs 
@gnufied does the uncertain mount fixes help with this?

**Comment by tedyu** [comment]
Would #89090 help in this case ?

**Comment by raunakkumar** [comment]
Probably not because we didn't get a nodePublish, we were stuck after nodeStage, so there would not be any nodeUnpublish

**Comment by tedyu** [comment]
Looks like we should do this:
```
diff --git a/pkg/volume/csi/csi_client.go b/pkg/volume/csi/csi_client.go
index c1d548c9f53..7bbb0326c9b 100644
--- a/pkg/volume/csi/csi_client.go
+++ b/pkg/volume/csi/csi_client.go
@@ -246,6 +246,11 @@ func (c *csiDriverClient) NodePublishVolume(
        }

        _, err = nodeClient.NodePublishVolume(ctx, req)
+       if err != nil {
+               if unstageErr := c.NodeUnstageVolume(ctx, volID, targetPath); unstageErr != nil {
+                       klog.Warningf("unable to unstage volume %s at %s", volID, targetPath)
+               }
+       }
        if err != nil && !isFinalError(err) {
                return volumetypes.NewUncertainProgressError(err.Error())
        }
```
@msau42 @gnufied 
If you agree, I can roll the above into #89090.

**Comment by jsafrane** [comment]
@raunakkumar, what Kubernetes version do you use? Was it filesystem or block volume? And can you please re-check with current master?

**Comment by raunakkumar** [comment]
We are using kubernetes 1.17 and the issue is with filesystem backed volume.
Any specific fix which went into master?

**Comment by gnufied** [comment]
Uncertain mount fix merged in https://github.com/kubernetes/kubernetes/pull/82492 should fix this one. Unfortunately this fix is not available in 1.17. Can you check with current master and see if you can still reproduce the bug?
